### PR TITLE
Feat 888 reset fetchers on yml changes

### PR DESCRIPTION
**Pull request by casscauley**
This creates a runner class that contains all the logic for parsing the agencies+headers yml files and creating the `Fetcher` instances. `Ticker.tick` triggers `Runner.tick`, which reloads the feeds by reloading the agencies+headers yml files. If the feeds are different, all fetchers are stopped and new fetchers are created.

## Testing
* I started by loading a simplified version of `agencies.yml`. 
* I then modified agencies.yml which caused the "Removing # old feeds and adding # new feeds" message to display in the terminal (it takes up to 20s for the next tick to occur)
* Same with headers.yml

## Notes:
* The Fetchers take 1 tick to fully stop/start so although there is no hiccup, there will be one more data point with the old feeds after the new feeds load.
* This doesn't actually do anything right now since the CI script still redeploys the container on every build. So #815 needs to be completed to fully utilize these changes.
* My second request makes a `get_headers_yml` and `get_agencies_yml` methods. Changing these to optionally pull from a bucket instead of the file system should be the only necessary change to this file to satisfy #815 and finish this process.

### CODE REVIEW

**Review by evansiroky — APPROVED** [review]
Code looks good, I had just one comment about how all fetchers get recreated instead of just the one affected, but that's probably fine and not blocking. 

Also, I'm just now noticing that this whole file is a little light on the code comments. If a little more code comments could be added especially for overall classes, that would be nice, but can be done in a separate effort.

**Review by lottspot — COMMENTED** [review]
I've got some additional changes coming based on this branch to address #815 -- should have them ready to merge Monday. Not blocking, but if we don't get around to merging this before then I'll add on top of this PR

File: services/gtfs-rt-archive/gtfs-rt-archive.py [hunk 1]
```diff
@@ -12,6 +12,97 @@
...
+            self.reset()
+
+    def reset(self):
+        fetchers = len(self.fetchers)
+        feeds = len(self.feeds)
+        self.logger.info(f"Removing {fetchers} old feeds and adding {feeds} new feeds")
+        for fetcher in self.fetchers:
+            fetcher.stop()
+
+        self.fetchers = []
```
[the remark this answers is not in the document]
Comment (chriscauley) [reply services/gtfs-rt-archive/gtfs-rt-archive.py:42]: I had to check that myself as well. It turns out in python `==` is a deep comparison.

**Comment by lottspot** [comment]
This PR is superseded by #999

**Pull request closed without merging**